src/utils/conversion.ts

```diff
@@ -3362,3 +3364,32 @@ export function agentGroupToPermissionGroup(
     }
   }
 }
+
+/**
+ * @hidden
+ */
+export async function portfolioNameToNumber(
+  rawIdentityId: IdentityId,
+  rawName: PortfolioName,
+  query: QueryableStorage<'promise'>
```

Comment:
I'd prefer passing in the Context so the signature looks more consistent and so a caller doesn't need to know about the destructing stuff.